### PR TITLE
[HIBIKE] heartbeat upgrade to use requests

### PR DESCRIPTION
**Pull request by dHutchings**
Smart devices now send packets asking for heartbeats, and get heartbeat responses back.

Can change the period between sending heartbeats.  Can change the time until it disables itself because no heartbeat responses.

### CODE REVIEW

File: hibike/lib/hibike/hibike_device.h [hunk 1]
```diff
@@ -32,7 +32,7 @@ typedef enum {
 #define LED_PIN 13
 
 extern hibike_uid_t UID;
-void hibike_setup(uint32_t _disable_latency);
+void  hibike_setup(uint32_t _disable_latency, uint32_t _heartbeatDelay);
```
Comment (kevinji) [inline hibike/lib/hibike/hibike_device.h]: Extra space got added here.

File: hibike/hibike_process.py [hunk 2]
```diff
@@ -115,11 +115,16 @@ def device_read_thread(index, ser, instructionQueue, errorQueue, stateQueue):
...
                 stateQueue.put(("device_values", [uid, params_and_values]))
-                instructionQueue.put(("heartResp", [uid]))
             else:
                 print("[HIBIKE] Port %s received data before enumerating!!!" % ser.port)
                 print("Telling it to shut up")
                 hm.send(ser, hm.make_subscription_request(1, [], 0))
+        elif message_type == hm.messageTypes["HeartBeatRequest"]:
+            if uid is not None:
+                instructionQueue.put(("heartResp", [uid]))
+
```
Comment (kevinji) [inline hibike/hibike_process.py]: Extra newlines here.

**Review by kevinji — COMMENTED** [review]
Just some minor spacing issues.

File: hibike/lib/hibike/hibike_device.h (hunk 1 above)
Comment (dHutchings) [reply hibike/lib/hibike/hibike_device.h]: fixed.

File: hibike/hibike_process.py (hunk 2 above)
Comment (dHutchings) [reply hibike/hibike_process.py]: Fixed

**Comment by dHutchings** [comment]
@mc10   Fixed the whitespace issues.  Thanks for reviewing!

Also fixed a merge into master that somehow got on this branch.  Just now committing actual changes for this feature.

**Comment by dHutchings** [comment]
That "Merge branch 'master' " was me fixing merge conflicts.  Dunno if that's the right thing to do...?

**Comment by kevinji** [comment]
That should be fine, since we're squashing and merging before any PR goes into master anyways.

File: hibike/devices/BatteryBuzzer/battery_buzzer.cpp [hunk 3]
```diff
@@ -28,7 +28,7 @@ void setup() {
   test_display();    //For production to check that the LED display is working.
   setup_sensing();
 
-  hibike_setup(500); // Time in milliseconds before timeout on heartbeat
+  hibike_setup(500, 100); //500 ms without heartbeat to disable, ask for heartbeats at 100 Hz.
```
Comment (yizow) [inline hibike/devices/BatteryBuzzer/battery_buzzer.cpp]: I recommend we tune down to 10 hz
100 hz is an order of magintude faster than the rest of the communication stack

**Review by yizow — CHANGES_REQUESTED** [review]

**Review by yizow — APPROVED** [review]

**Review by yizow — APPROVED** [review]

**Pull request merged**